### PR TITLE
Unify titles in rustdoc book doc attributes chapter

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
As discussed in https://github.com/rust-lang/rust/pull/90339.

I wasn't able to find out where the link to the titles was used so let's see if the CI fails. :)

r? @camelid

### CODE REVIEW

**Review by camelid — COMMENTED** [review]
r=me with the following comments addressed and the error message I mentioned above updated (you'll have to bless tests as well)

Please make sure to use `r=camelid` rather than `r+` :)

**Comment by GuillaumeGomez** [comment]
Updated! I'll let you handle the r+ this time. ;)

File: compiler/rustc_passes/src/check_attr.rs [hunk 1]
```diff
@@ -768,7 +768,7 @@ impl CheckAttrVisitor<'tcx> {
                             "not a `use` item",
                         );
                     }
-                    err.note("read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#docno_inlinedocinline for more information")
+                    err.note("read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#inline_and_no_inline for more information")
```
Comment (camelid) [inline compiler/rustc_passes/src/check_attr.rs:771]: I think this link is not quite correct. It should be:

```suggestion
                    err.note("read https://doc.rust-lang.org/nightly/rustdoc/the-doc-attribute.html#inline-and-no_inline for more information")
```

Comment (GuillaumeGomez) [reply compiler/rustc_passes/src/check_attr.rs]: Arf, copied the link I created and forgot to regen the book... Great catch!

**Comment by camelid** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**